### PR TITLE
feat: add non-root user

### PR DESCRIPTION
**Pull request by hoffa**
### Description

Adds non-root `sam` to all build images. Running as `root` isn't often necessary.

Tested with:

```shell
SAM_CLI_VERSION=1.21.0 make build
ulimit -n 1000
SAM_CLI_VERSION=1.21.0 make test
```

Had to increase open-file limit as `ulimit -n` showed 256. Might be bug leaking in testing code or expected; haven't had time to look into it too deep.

Also had to update mirror as [current one is down](https://www.apache.org/mirrors/). New one also uses HTTPS.

### CODE REVIEW

File: build-image-src/Dockerfile-java8-al2 [hunk 1]
```diff
@@ -58,9 +58,11 @@ ENV JAVA_HOME="/usr/lib/jvm/java-1.8.0-openjdk"
 
 RUN mkdir /usr/local/gradle && curl -L -o gradle.zip https://downloads.gradle-dn.com/distributions/gradle-6.2.2-bin.zip && \
   unzip -d /usr/local/gradle gradle.zip && rm gradle.zip && mkdir /usr/local/maven && \
-  curl -L http://mirror.cc.columbia.edu/pub/software/apache/maven/maven-3/3.6.3/binaries/apache-maven-3.6.3-bin.tar.gz | \
+  curl -L https://mirrors.ocf.berkeley.edu/apache/maven/maven-3/3.6.3/binaries/apache-maven-3.6.3-bin.tar.gz | \
```
Comment (prenx4x) [inline build-image-src/Dockerfile-java8-al2:61]: Why this change?

File: tests/build_image_base_test.py [hunk 2]
```diff
@@ -66,6 +66,17 @@ def test_common_packages(self):
...
+    def test_non_root_user(self):
+        """
+        Test using non-root `sam` user
+        """
+        output = (
+            self.client.containers.run(image=self.image, user="sam", command="whoami")
+            .decode()
+            .strip()
+        )
+        self.assertEqual(output, "sam")
```
Comment (prenx4x) [inline tests/build_image_base_test.py]: Do we need a test that tries to run a command but fails without root permissions?

File: build-image-src/Dockerfile-ruby27 [hunk 3]
```diff
@@ -50,4 +50,6 @@ RUN pip3 install wheel
 
 ENV LANG=en_US.UTF-8
 
+RUN /sbin/useradd sam
```
Comment (prenx4x) [inline build-image-src/Dockerfile-ruby27]: does this user have sudo priviledges?

File: tests/build_image_base_test.py (hunk 2 above)
Comment (hoffa) [reply tests/build_image_base_test.py]: Good catch, I changed so it checks that UID is not 0 (which is [always 0 for the root user](https://superuser.com/questions/626843/does-the-root-account-always-have-uid-gid-0)).

File: build-image-src/Dockerfile-ruby27 (hunk 3 above)
Comment (hoffa) [reply build-image-src/Dockerfile-ruby27]: No, that would require passing `-r`. This is checked by the updated test.

File: build-image-src/Dockerfile-java8-al2 (hunk 1 above)
Comment (hoffa) [reply build-image-src/Dockerfile-java8-al2:61]: See link in summary; the mirror is down. (Plus this adds TLS)

**Review by prenx4x — APPROVED** [review]

File: build-image-src/Dockerfile-ruby27 (hunk 3 above)
Comment (hoffa) [reply build-image-src/Dockerfile-ruby27]: Switched to using explicit arguments.

File: build-image-src/Dockerfile-java11 [hunk 4]
```diff
@@ -65,9 +65,11 @@ ENV JAVA_HOME="/var/lang"
 
 RUN mkdir /usr/local/gradle && curl -L -o gradle.zip https://downloads.gradle-dn.com/distributions/gradle-6.2.2-bin.zip && \
   unzip -d /usr/local/gradle gradle.zip && rm gradle.zip && mkdir /usr/local/maven && \
-  curl -L http://mirror.cc.columbia.edu/pub/software/apache/maven/maven-3/3.6.3/binaries/apache-maven-3.6.3-bin.tar.gz | \
+  curl -L https://mirrors.ocf.berkeley.edu/apache/maven/maven-3/3.6.3/binaries/apache-maven-3.6.3-bin.tar.gz | \
```
Comment (aahung) [inline build-image-src/Dockerfile-java11:68]: does downloads.apache.org work? It feels more official than the berkeley mirror

Comment (hoffa) [reply build-image-src/Dockerfile-java11:68]: Didn't use that as it's called "Backup" on https://www.apache.org/mirrors/ 🤔

Comment (aahung) [reply build-image-src/Dockerfile-java11:68]: oh I see, that's fair, thx!

**Review by aahung — APPROVED** [review]

File: build-image-src/Dockerfile-java11 [hunk 5]
```diff
@@ -65,9 +65,11 @@ ENV JAVA_HOME="/var/lang"
...
   tar -zx -C /usr/local/maven
 
 ENV PATH="/usr/local/gradle/gradle-6.2.2/bin:/usr/local/maven/apache-maven-3.6.3/bin:${PATH}"
 
+RUN /usr/sbin/useradd --uid 1000 --user-group --shell /bin/bash --create-home sam
```
Comment (sriram-mv) [inline build-image-src/Dockerfile-java11]: nit and non blocking, could we add `--comment` too.

**Review by sriram-mv — APPROVED** [review]
